### ISSUE TITLE
NotNullConstraintViolationException in Bolt 2 > 3 upgraded installs

### ISSUE THREAD

**Issue by peterboorsma** [bolt/bolt#6498]
After upgrading a Bolt 2 MYSQL database to Bolt 3, a user are unable to create new records if the contenttype has image or video fields.

Those field types need a NULL value (I think) but this value is not assigned in the update process.

Tested with Bolt 3.2.9
PHP 7.0.13
MySQL version 5.6.34

```
Uncaught Exception: NotNullConstraintViolationException

NotNullConstraintViolationException in AbstractMySQLDriver.php line 112: 
An exception occurred while executing 'INSERT INTO bolt_nieuws_nl (slug, datecreated, datechanged, datepublish, datedepublish, username, ownerid, status, templatefields, title, teaser, broodtekst, image, video) VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params ["test", "2017-03-20 16:55:52", "2017-03-20 16:55:52", "2017-03-20 16:55:48", null, "", "1", "published", "[]", "test", "", "", null, null]:

SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'image' cannot be null

```

**Comment by GawainLynch** [comment]
The [schema column change exclusion is this line](https://github.com/bolt/bolt/blob/release/3.2/src/Storage/Database/Schema/Comparison/MySql.php#L24]. I commented mine out, ran DB update, and didn't seem to strike any problems post that (DB columns updated correctly).

There was a reason we had that line, but I can't remember what it was … Buller?

**Comment by Finn10111** [comment]
I am using Postgresql and commented out the line you mentioned (in the PostgreSql.php file). A database update gives me the following error:

> Modifications made to the database:
>  - Table `bolt_users` is not the correct schema: invalid column `stack`, invalid column `roles`
> Your database is now up to date.

Also the NotNullContrainViolationException still occurs.

**Comment by GawainLynch** [comment]
This all changes quite a bit in #6940 … leaving that here for cress-reference when that lands shortly

**Comment by GawainLynch** [comment]
Will have been fixed in #6940
